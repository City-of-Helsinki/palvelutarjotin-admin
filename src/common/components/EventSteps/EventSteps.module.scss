.formSteps {
  --underline-size: 5px;

  display: inline-grid;
  grid-template-columns: auto auto auto;
  grid-gap: 3rem;
  font-size: 18px;
  font-weight: 700;

  > div {
    padding-bottom: calc(var(--spacing-xs) * 0.5);
  }

  .active {
    border-bottom: var(--underline-size) solid black;
  }
}
