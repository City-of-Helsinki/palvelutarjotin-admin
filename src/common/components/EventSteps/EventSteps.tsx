import classNames from 'classnames';
import * as React from 'react';
import { useTranslation } from 'react-i18next';

import SrOnly from '../SrOnly/SrOnly';
import styles from './EventSteps.module.scss';

interface EventStepsProps {
  step: 1 | 2 | 3;
}

const EventSteps: React.FC<EventStepsProps> = ({ step }) => {
  const { t } = useTranslation();
  const stepTitles = [
    t('createEvent.stepTitles.titleBasicInfo'),
    t('createEvent.stepTitles.titleOccurrences'),
    t('createEvent.stepTitles.titlePublish'),
  ];

  return (
    <>
      <div className={styles.formSteps} aria-hidden="true">
        {stepTitles.map((title, i) => (
          <div
            key={title}
            className={classNames({ [styles.active]: i + 1 === step })}
          >
            {`${i + 1}. ${title}`}
          </div>
        ))}
      </div>
      <SrOnly>
        <p>{`${step}. ${stepTitles[step - 1]}`}</p>
      </SrOnly>
    </>
  );
};

export default EventSteps;
