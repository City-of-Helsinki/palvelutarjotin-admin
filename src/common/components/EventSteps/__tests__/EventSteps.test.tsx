import { render, screen } from '@testing-library/react';
import React from 'react';

import EventSteps from '../EventSteps';

it('matches snapshot', () => {
  const { container } = render(<EventSteps step={1} />);

  expect(container).toMatchSnapshot();
});

it('renders correct title', () => {
  const { rerender } = render(<EventSteps step={1} />);

  expect(screen.getAllByText('1. Perustiedot')).toHaveLength(2);
  expect(screen.getAllByText('1. Perustiedot')[0]).toHaveClass('active');

  rerender(<EventSteps step={2} />);

  expect(screen.getAllByText('2. Tapahtuma-ajat')).toHaveLength(2);
  expect(screen.getAllByText('2. Tapahtuma-ajat')[0]).toHaveClass('active');

  rerender(<EventSteps step={3} />);

  expect(screen.getAllByText('3. Julkaise')).toHaveLength(2);
  expect(screen.getAllByText('3. Julkaise')[0]).toHaveClass('active');
});
