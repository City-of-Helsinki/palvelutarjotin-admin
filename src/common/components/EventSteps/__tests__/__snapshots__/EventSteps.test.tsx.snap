// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    aria-hidden="true"
    class="formSteps"
  >
    <div
      class="active"
    >
      1. Perustiedot
    </div>
    <div
      class=""
    >
      2. Tapahtuma-ajat
    </div>
    <div
      class=""
    >
      3. Julkaise
    </div>
  </div>
  <div
    class="srOnly"
  >
    <p>
      1. Perustiedot
    </p>
  </div>
</div>
`;
